### PR TITLE
Added test to ensure GetSignup endpoint updates IdentityClaims

### PR DESCRIPTION
**Pull request by sbryzak**
Related PR: https://github.com/codeready-toolchain/registration-service/pull/359

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]
Looks good. Just one linter issue to address.

File: test/e2e/usersignup_test.go [hunk 1]
```diff
@@ -238,6 +242,76 @@ func (s *userSignupIntegrationTest) TestUserIDAndAccountIDClaimsPropagated() {
...
+	req.Header.Set("Authorization", "Bearer "+token)
+	req.Header.Set("content-type", "application/json")
+
+	req.Close = true
+	resp, err := httpClient.Do(req) // nolint:bodyclose // see `defer Close(t, resp)`
+	require.NoError(s.T(), err)
+	b, err := io.ReadAll(resp.Body)
+	require.Equal(s.T(), http.StatusOK, resp.StatusCode, "unexpected status, body: %s", string(b))
+
+	defer Close(s.T(), resp)
```
Comment (mfrancisc) [inline test/e2e/usersignup_test.go:297]: minor thing - but probably you could use `InvokeEnpoint` utility here. 
See: https://github.com/codeready-toolchain/toolchain-e2e/blob/8eeaeaf8f0c07e64a65df8dbba596fecfb28dd1f/testsupport/regsvc.go#L14

Comment (sbryzak) [reply test/e2e/usersignup_test.go]: Great suggestion, thanks Francisc

**Pull request merged**